cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -245,23 +249,6 @@ func isServiceActive(name string) (bool, error) {
 // TODO: https://github.com/kubernetes/kubeadm/issues/2376
 const cgroupDriverSystemd = "systemd"
 
-// MutateCgroupDriver can be called to set the KubeletConfiguration cgroup driver to systemd.
-// Currently this cannot be as part of Default() because the function is called for
-// upgrades too, which can break existing nodes after a kubelet restart.
-// TODO: https://github.com/kubernetes/kubeadm/issues/2376
-func MutateCgroupDriver(cfg *kubeadmapi.ClusterConfiguration) {
-	cc, k, err := getKubeletConfig(cfg)
-	if err != nil {
-		klog.Warningf(err.Error())
-		return
-	}
-	if len(k.CgroupDriver) == 0 {
-		klog.V(1).Infof("setting the KubeletConfiguration cgroupDriver to %q", cgroupDriverSystemd)
-		k.CgroupDriver = cgroupDriverSystemd
-		cc.Set(k)
-	}
-}
-
 // WarnCgroupDriver prints a warning in case the user is not explicit
```

Comment:
we should remove this function too it's called from "upgrade plan".

Reply:
Updated